core/metacling/src/TClingCallbacks.cxx

```diff
@@ -919,7 +922,48 @@ void TClingCallbacks::UnlockCompilationDuringUserCodeExecution(void *StateInfo)
    TCling__UnlockCompilationDuringUserCodeExecution(StateInfo);
 }
 
-static bool shouldIgnore(llvm::StringRef FileName) {
+static bool shouldIgnore(const std::string& FileName,
+                         cling::DynamicLibraryManager* dyLibManager) {
+   if (llvm::sys::fs::is_directory(FileName))
+     return true;
+
+   if (!cling::DynamicLibraryManager::isSharedLibrary(FileName))
+     return true;
+
+   // No need to check linked libraries, as this function is only invoked
+   // for symbols that cannot be found (neither by dlsym nor in the JIT).
+   if (dyLibManager->isLibraryLoaded(FileName.c_str()))
+      return true;
+
+
+   auto ObjF = llvm::object::ObjectFile::createObjectFile(FileName);
+   if (!ObjF) {
+      if (gDebug > 1)
+         ROOT::TMetaUtils::Warning("[DyLD]", "Failed to read object file %s",
+                                   FileName.c_str());
+      return true;
+   }
+
+   llvm::object::ObjectFile *file = ObjF.get().getBinary();
+
+   if (isa<llvm::object::ELFObjectFileBase>(*file)) {
+      for (auto S : file->sections()) {
+         StringRef name;
+         S.getName(name);
+         if (name == ".text") {
+            // Check if the library has only debug symbols, usually when
+            // stripped with objcopy --only-keep-debug
+            auto SecRef = static_cast<llvm::object::ELFSectionRef&>(S);
```

Comment:
Are the bases for these test documented somewhere (a reference would then be helpful) or discovered (i.e. heuristic)?

Reply:
We created a library and looked. The best reference is the manual for `objcopy` which does not have the information we need.

What do you mean by bases: `ELFSectionRef`?

Reply:
By "bases" I meant "source of information/knowledge", so given your answer you may want to add a comment akin to "Test extracted from looking at a library created with binutils X.Y.Z."